src/editor.rs

```diff
@@ -655,13 +655,14 @@ impl Editor {
     ///
     /// Will Return `Err` if any error occur.
     pub fn run(&mut self, file_name: Option<String>) -> Result<(), Error> {
-        if let Some(path) = file_name.as_ref().map(Path::new) {
-            self.select_syntax_highlight(path)?;
-            self.load(path)?;
+        if let Some(path) = file_name.as_ref().map(sys::path) {
+            self.select_syntax_highlight(path.as_path())?;
+            self.load(path.as_path())?;
+            self.file_name = Some(path.to_string_lossy().to_string());
         } else {
             self.rows.push(Row::new(Vec::new()));
+            self.file_name = file_name;
```

Comment:
I think this would be clearer:
```suggestion
            self.file_name = None;
```

Reply:
updated - it is indeed clearer